packages/server/src/workers/post-deploy-migration.ts

```diff
@@ -78,7 +79,27 @@ export async function jobProcessor(job: Job<PostDeployJobData>): Promise<void> {
     version: migrationNumber,
   });
 
-  const migration = getPostDeployMigration(migrationNumber);
+  let migration: PostDeployMigration;
+  try {
+    migration = getPostDeployMigration(migrationNumber);
+  } catch (err: any) {
+    if (err instanceof MigrationDefinitionNotFoundError) {
+      const delayMs = 60_000;
+      globalLogger.info(
+        `${PostDeployMigrationQueueName} processor delaying job since migration definition was not found on this worker`,
+        {
+          jobId: job.id,
+          delayMs,
+          ...getJobDataLoggingFields(job),
+          version: migrationNumber,
+        }
+      );
+      await job.moveToDelayed(Date.now() + delayMs, job.token);
+      throw new DelayedError('Post-deploy migration delayed since migration definition was not found on this worker');
```

Comment:
Interesting model for delaying jobs, calling both a method on the job and throwing an error. TIL

https://docs.bullmq.io/patterns/process-step-jobs#delaying